code/modules/clothing/masks/gasmask.dm

```diff
@@ -109,35 +109,51 @@
 
 /obj/item/clothing/mask/gas/clown_hat
 	name = "clown wig and mask"
-	desc = "A true prankster's facial attire. A clown is incomplete without his wig and mask."
+	desc = "A true prankster's facial attire. A clown is incomplete without his wig and mask. Its form can be changed by using it in your hand."
 	icon_state = "clown"
 	item_state = "clown_hat"
 	flags = BLOCK_GAS_SMOKE_EFFECT | AIRTIGHT | BLOCKHAIR
 	resistance_flags = FLAMMABLE
 	dog_fashion = /datum/dog_fashion/head/clown
 
-/obj/item/clothing/mask/gas/clown_hat/attack_self(mob/user)
-
+/obj/item/clothing/mask/gas/clown_hat/attack_self(mob/living/user)
 	var/mob/M = usr
-	var/list/options = list()
-	options["True Form"] = "clown"
-	options["The Feminist"] = "sexyclown"
-	options["The Madman"] = "joker"
-	options["The Rainbow Color"] ="rainbow"
-
-	var/choice = input(M,"To what form do you wish to Morph this mask?","Morph Mask") in options
-
-	if(src && choice && !M.stat && in_range(M,src))
-		icon_state = options[choice]
-		to_chat(M, "Your Clown Mask has now morphed into [choice], all praise the Honk Mother!")
+	var/list/mask_type = list("True Form" = /obj/item/clothing/mask/gas/clown_hat,
+							"The Feminist" = /obj/item/clothing/mask/gas/clown_hat/sexy,
+							"The Madman" = /obj/item/clothing/mask/gas/clown_hat/joker,
+							"The Rainbow Color" = /obj/item/clothing/mask/gas/clown_hat/rainbow)
+	var/list/mask_icons = list("True Form" = image(icon = 'icons/obj/clothing/masks.dmi', icon_state = "clown"),
+							"The Feminist" = image(icon = 'icons/obj/clothing/masks.dmi', icon_state = "sexyclown"),
+							"The Madman" = image(icon = 'icons/obj/clothing/masks.dmi', icon_state = "joker"),
+							"The Rainbow Color" = image(icon = 'icons/obj/clothing/masks.dmi', icon_state = "rainbow"))
+	var/mask_choice = show_radial_menu(user, src, mask_icons)
+	var/picked_mask = mask_type[mask_choice]
+
+	if(src && picked_mask && !M.stat && in_range(M,src))
```

Comment:
```suggestion
	if(QDELETED(src) || !picked_mask)
		return
	if(M.stat || !in_range(M, src))
	    return
```
these need to be early returns, instead of checking to see if `src` exists, you should check to see if it has been qdel'd